scripts/core/editor3/components/Editor3Component.tsx

```diff
@@ -428,6 +433,12 @@ export class Editor3Component extends React.Component<IProps, IState> {
             return 'handled';
         }
 
+        if (this.props.limit && this.props.limitBehavior === 'limit') {
+            if (preventInputWhenLimitIsPassed(editorState, chars, this.props.limit)) {
```

Comment:
The limit should apply to all types changes to editor state, not only to keystrokes. For example, it doesn't apply the limit when pasting.

Reply:
True, right now if you paste some text that overflows the field it does highlight it in red (instead of chopping it). I thought this was more user friendly, otherwise you would lose the rest of the text when pasting, not sure if that's a good UX

Reply:
> otherwise you would lose the rest of the text when pasting, not sure if that's a good UX

That's how `maxlength` works with native input/textarea so I guess it'd be good enough here as well. If requirements allow it, I guess we could keep highlighting too.

Reply:
Ok, I talked with Ana and they want the behavior we currently have. Allow pasting (highlighting overflow) and not chopping the overflow when switching options